### PR TITLE
VB-1208: Use new endpoints for update journey when selecting date/time

### PR DESCRIPTION
**Pull request by tpmcgowan**
Changes in this PR:
* Use new endpoints when selecting date/time for update journey so that the same visit reference is preserved
* Update tests for select date/time to cover both book and update journeys

The new flow when selecting date/time is:

**Booking journey**
- first time confirming date/time => 'reserve new visit'; get reference and applicationReference; status = RESERVED
- nth time (before booking) => 'change reserved visit'; status stays as RESERVED

**Update visit journey**
- first time confirming date/time => 'change booked visit'; get new applicationReference, with status of RESERVED (for new date/time or visitRestriction) or CHANGING (for same date/time or visitRestriction)
- nth time (before booking) => 'change reserved visit'; status RESERVED or CHANGING depending on whether date/time or restriction changed

### CODE REVIEW

File: server/@types/bapv.d.ts [hunk 1]
```diff
@@ -196,7 +196,6 @@ export type VisitSessionData = {
   }
   applicationReference?: string
   visitReference?: string
-  previousVisitReference?: string
```
Comment (tpmcgowan) [inline server/@types/bapv.d.ts:199]: Most of the changes in this PR are actually the result of removing this 🙂 ! No longer needed as `visitReference` stays the same and the work is done by `applicationReference`.

Comment (tpmcgowan) [inline server/@types/bapv.d.ts:199]: Most of the changes in this PR are actually the result of removing this 🙂 ! No longer needed as `visitReference` stays the same and the work is done by `applicationReference`.

File: server/routes/visitJourney/checkYourBooking.ts [hunk 2]
```diff
@@ -74,27 +73,6 @@ export default class CheckYourBooking {
         operationId: res.locals.appInsightsOperationId,
       })
 
-      if (isUpdate) {
```
Comment (tpmcgowan) [inline server/routes/visitJourney/checkYourBooking.ts:77]: The actual 'booking' stage at the end is now the same for first-time booking or updating existing booking.

File: server/routes/visitJourney/dateAndTime.ts [hunk 3]
```diff
@@ -107,11 +107,30 @@ export default class DateAndTime {
 
     visitSessionData.visit = getSelectedSlot(req.session.slotsList, req.body['visit-date-and-time'])
 
+    /* Flow here is:
```
Comment (psoleckimoj) [inline server/routes/visitJourney/dateAndTime.ts:110]: Should probably be in the `README.md`?

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: server/routes/visitJourney/dateAndTime.ts (hunk 3 above)
Comment (tpmcgowan) [reply server/routes/visitJourney/dateAndTime.ts:110]: Yeah good idea 👍 . I'll move it over.

**Review by hutcheonb-moj — APPROVED** [review]
LGTM

**Pull request merged**